### PR TITLE
Optimized parser

### PR DESCRIPTION
**Pull request by At-sushi**
An optimization for the parser class.

### CODE REVIEW

File: src/parser.ts [hunk 1]
```diff
@@ -24,59 +24,46 @@ class Parser extends EventEmitter {
...
     chunk = this.message
 
-    const size: number = chunk.length
-    let start: number = 0
-    let offset: number = 0
-    let curr: string | undefined
-    let next: string | undefined
+    let chunk_split: Array<string> = chunk.split('\n\n')
+    
+    for (var i = 0; i < chunk_split.length - 1; i++) {
```
Comment (h3poteto) [inline src/parser.ts]: __[TSLint]__ <sub>posted by Sider</sub>

Forbidden 'var' keyword, use 'let' or 'const' instead (no-var-keyword)

File: src/parser.ts [hunk 2]
```diff
@@ -24,59 +24,46 @@ class Parser extends EventEmitter {
     this.message += chunk
     chunk = this.message
 
-    const size: number = chunk.length
-    let start: number = 0
-    let offset: number = 0
-    let curr: string | undefined
-    let next: string | undefined
+    let chunk_split: Array<string> = chunk.split('\n\n')
+
```
Comment (h3poteto) [inline src/parser.ts]: __[TSLint]__ <sub>posted by Sider</sub>

trailing whitespace (no-trailing-whitespace)

File: src/parser.ts [hunk 3]
```diff
@@ -24,59 +24,46 @@ class Parser extends EventEmitter {
...
-    const size: number = chunk.length
-    let start: number = 0
-    let offset: number = 0
-    let curr: string | undefined
-    let next: string | undefined
+    let chunk_split: Array<string> = chunk.split('\n\n')
+    
+    for (var i = 0; i < chunk_split.length - 1; i++) {
+      const piece: string = chunk_split[i]
+
```
Comment (h3poteto) [inline src/parser.ts]: __[TSLint]__ <sub>posted by Sider</sub>

trailing whitespace (no-trailing-whitespace)

**Comment by h3poteto** [comment]
What is the purpose of this pull request? Refactoring? Or bug fix?

**Comment by At-sushi** [comment]
I oriented for cpu resource reduction, but the test has shown that this has no efficiency.
So, I cancel this request.

Thanks for your mention.

**Pull request closed without merging**